pep-0480.txt

```diff
@@ -373,46 +388,78 @@ across multiple machines of each developer.
+solution is to use ephemeral keys, where the private key is only kept in memory
+(and not stored on disk) for as long as needed to produce the signature and is
+then safely discarded. This removes any long-term challenge of protecting the
+key by doing away with the key in the first place.
+
+`Sigstore`__ provides a way to create and use such ephemeral keys and maintain
+a chain of trust between the signature made on a distribution and the publicly
+available information stored on its transparency log. It uses certificates
+issued by ts certificate authority (Fulcio) to prove a signing key was valid
+when a signature was made. In-depth discussion of sigstore's architecture is
+beyond the scope of this document and for more information one may refer to
+https://sigstore.dev
+
+Ephemeral keys tie to an OpenID Connect identity, which is portable across
+multiple computers and does not require any key sharing. It also relies on an
+account a developer already uses, removing the issue of requiring a further
+password to remember.
+
+Optionally, sigstore also allows for manually managed keypairs. Its cosign tool
+can generate keys using the Ed25519 signature scheme.
+
+__ https://github.com/sigstore/
+
 The following outlines an automated signing solution that a new developer MAY
 follow to upload a distribution to PyPI:
 
 1.  Register a PyPI project.
-2.  Enter a secondary password (independent of the PyPI user account password).
-3.  Optional: Add a new identity to the developer's PyPI user account from a
-    second machine (after a password prompt).
-4.  Upload project.
+2.  Register an OIDC identity for the project.
+  2.1 Alternative: register a public key.
+  2.2 Optional: Add a new identity to the developer's PyPI user account from a
+    second machine.
+3.  Upload project.
 
 Step 1 is the normal procedure followed by developers to `register a PyPI
 project`__.
 
 __ https://pypi.python.org/pypi?:action=register_form
 
-Step 2 generates an encrypted key file (private), uploads an Ed25519 public key
-to PyPI, and signs the TUF metadata that is generated for the distribution.
+Step 2 registers an OIDC identity to PyPI, generates an ephemeral key and signs the
+TUF metadata that is generated for the distribution.
 
-Optionally adding a new identity from a second machine, by simply entering a
-password, in step 3 also generates an encrypted private key file and uploads an
-Ed25519 public key to PyPI.  Separate identities MAY be created to allow a
-developer, or other project maintainers, to sign releases on multiple machines.
-An existing verified identity (its public key is contained in project metadata
-or has been uploaded to PyPI) signs for new identities.  By default, project
-metadata has a signature threshold of "1" and other verified identities may
-create new releases to satisfy the threshold.
+Optionally, a long-lived keypair MAY be generated instead (where the private
```

Comment:
Agree with Justin and Dustin: we should talk about one option at a time, not both.